src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,855 +1,245 @@
 ---
-<Fragment set:html={html} />
+<Heading title="My First Post" date="09 Aug 2022" />
 ```
 
-:::note
-This is for advanced use cases! In most circumstances, it is simpler to render slot contents with [the `<slot />` element](/en/basics/astro-components/#slots).
-:::
+<ReadMore>Learn more about how [Markdown and MDX layouts](/en/guides/markdown-content/#frontmatter-layout-property) handle props.</ReadMore>
+
+
+The `props` object also contains any `props` passed from `getStaticPaths()` when rendering static routes.
+
+<Tabs>
+  <TabItem label="Astro.props">
+    ```astro title="src/pages/posts/[id].astro" "props:" {11}
+    ---
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' }, props: { author: 'Blu' } },
+        { params: { id: '2' }, props: { author: 'Erika' } },
+        { params: { id: '3' }, props: { author: 'Matthew' } }
+      ];
+    }
+
+    const { id } = Astro.params;
+    const { author } = Astro.props;
+    ---
+    ```
+  </TabItem>
+  <TabItem label="context.props">
+    ```ts title="src/pages/posts/[id].json.ts" "props:" {11-14}
+    import type { APIContext } from 'astro';
+
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' }, props: { author: 'Blu' } },
+        { params: { id: '2' }, props: { author: 'Erika' } },
+        { params: { id: '3' }, props: { author: 'Matthew' } }
+      ];
+    }
+
+    export function GET({ props }: APIContext) {
+      return new Response(
+        JSON.stringify({ author: props.author }),
+      );
+    }
+    ```
+  </TabItem>
+
+</Tabs>
+See also: [Data Passing with `props`](/en/reference/routing-reference/#data-passing-with-props)
 
-`Astro.slots.render()` optionally accepts a second argument: an array of parameters that will be forwarded to any function children. This can be useful for custom utility components.
-
-For example, this `<Shout />` component converts its `message` prop to uppercase and passes it to the default slot:
+### `params`
 
-```astro title="src/components/Shout.astro" "await Astro.slots.render('default', [message])"
+`params` is an object containing the values of dynamic route segments matched for a route request. Its keys must match the [dynamic route parameters](/en/guides/routing/#dynamic-routes) from the page file or endpoint route.
+
+In static builds, this will be the `params` returned by `getStaticPaths()` used for prerendering [dynamic routes](/en/guides/routing/#dynamic-routes):
+
+<Tabs>
+  <TabItem label="Astro.params">
+    ```astro title="src/pages/posts/[id].astro" {9} "params:"
+    ---
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' } },
+        { params: { id: '2' } },
+        { params: { id: '3' } }
+      ];
+    }
+    const { id } = Astro.params;
+    ---
+    <h1>{id}</h1>
+    ```
+  </TabItem>
+  <TabItem label="context.params">
+    ```ts title="src/pages/posts/[id].json.ts" "params:" {11-14}
```

Comment:
nit: same as above, I think the closing brace should be included:
```suggestion
    ```ts title="src/pages/posts/[id].json.ts" "params:" {11-15}
```